src/fetchTileData.ts

```diff
@@ -0,0 +1,60 @@
+// Adapted from https://github.com/perliedman/node-hgt/blob/master/src/hgt.js
+
+import fnv1a from '@sindresorhus/fnv1a';
+import {LRUCache} from 'lru-cache';
+
+// Generic type for a function that somehow fetches a tile and produces a `Buffer`, e.g.,
+// `s3Fetcher` fetches a tile from an S3 bucket.
+type Fetcher<T extends unknown[]> = (...args: T) => Promise<Buffer>;
+
+// Thrown by Fetchers to indicate a missing tile, which may (e.g. elevation) or may not be expected.
+export const TILE_MISSING = Symbol();
+
+// Generic error thrown by Fetchers to that signal something went wrong.
+export const BAD_TILE = Symbol();
+
+// ...could signal other Fetcher error conditions, e.g. bad permissions, network failure, etc.
+
+
+const cache = new LRUCache<bigint, Buffer>({
+    sizeCalculation: (buffer) => buffer.length, // bytes
+    maxSize: 500_000_000, // 500 MB, specified in bytes because `sizeCalculation` returns bytes
+});
+
+const pending = new Map<bigint, Promise<Buffer>>();
+
+const missing = new Set<bigint>();
+
+export async function fetchTileData<T extends unknown[]>(fetcher: Fetcher<T>, ...args: T) {
+    const hash = fnv1a(JSON.stringify(args));
+
+    // If the LRU cache has it, return it
+    if (cache.has(hash)) return cache.get(hash)!;
+
+    // If we're currently fetching it, return the promise
+    if (pending.has(hash)) return pending.get(hash)!;
+
+    // If we know we can't get it, don't bother trying again
+    if (missing.has(hash)) throw TILE_MISSING;
```

Comment:
do we want this to grow unbounded? should we make `missing` a separate LRU cache?

Reply:
Possibly. 

In practice, the theoretical maximum number of items in the set for HGT tiles is 32,400 since there is only one tile per whole-degree lng/lat pair. Even though it says `bigint`, we're using the 32-big version of fnv, so the data isn't really very big. In practice, figure about 70% of elevation tiles are missing (% ocean coverage).

If we add other types of data, like cell coverage, where we have many tiles at different zoom levels, then we might want to limit the size.